pkg/secretlock/local/local_secret_lock.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package local
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"encoding/base64"
+	"io"
+	"os"
+
+	"github.com/google/tink/go/subtle/random"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+)
+
+// Lock is a secret lock service responsible for encrypting keys using a master key stored as environment variable.
+type Lock struct {
+	aead cipher.AEAD
+}
+
+// NewService creates a new instance of a local secret lock service for a master key in path.
+// the master key should be an AES key with size either 16, 24, or 32 bytes to select AES-128, AES-192, or AES-256.
+func NewService(path string) (secretlock.Service, error) {
+	masterKeyFile, err := os.OpenFile(path, os.O_RDONLY, 0600)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewServiceFromReader(masterKeyFile)
+}
+
+// NewServiceFromReader creates a new instance of local secret lock service for a master key reader.
+// the master key should be an AES key with size either 16, 24, or 32 bytes to select AES-128, AES-192, or AES-256.
+func NewServiceFromReader(masterKeyReader io.Reader) (secretlock.Service, error) {
```

Comment:
Could we enforce key size? That is, return an error if the key provided is too short for the security policy?

Reply:
the new API generates the key according the cipher... the only thing the user is required to pass is a passphrase. There's no need to expose a key size option to the user.